### PR TITLE
feature(kamel): Validate --context when given

### PR DESCRIPTION
**Pull request by rhuss**

### CODE REVIEW

File: pkg/cmd/install.go [hunk 1]
```diff
@@ -230,3 +235,43 @@ func (o *installCmdOptions) waitForPlatformReady(platform *v1alpha1.IntegrationP
...
+	schema := runtime.NewScheme()
+	if err := apis.AddToScheme(schema); err != nil {
+		return err
+	}
+
+	if o.contexts == nil {
+		return nil
+	}
+	for _, context := range o.contexts {
+		err := errorIfContextIsNotAvailable(schema, context)
```
Comment (lburgazzoli) [inline pkg/cmd/install.go]: This does not take into account the "none" context

File: deploy/resources.go [hunk 2]
```diff
@@ -2797,6 +2797,29 @@ artifacts:
...
+
+      from('timer:groovy?period=1s')
+          .routeId('groovy')
+          .setBody()
+              .constant('Hello Camel K!')
+          .process {
+              it.in.headers['RandomValue'] = rnd.nextInt()
+          }
+          .to('log:info?showHeaders=true')
+    name: routes.groovy
```
Comment (lburgazzoli) [inline deploy/resources.go]: May this be changed by the fact that your pr is created from an old commit ?

File: pkg/cmd/install.go (hunk 1 above)
Comment (rhuss) [reply pkg/cmd/install.go]: Valid point. I added a check for "none", too now so that is a valid input.

File: deploy/resources.go (hunk 2 above)
Comment (rhuss) [reply deploy/resources.go]: Not sure. I added a sorting option to `ls` when creating that file previously, but it seems that the PR has not been taken over. Actually it then depends on the way how ls works on different systems how resources are picked up. My ls uses alphabetical sort order by default, so orders 'cr-example.yml' after 'camel-catalog-2.23.1.yaml' and before 'crd-camel-catalog.yaml'.

In this file frommaster 'cr-example.yml' comes right before 'operator-deployment-kubernetes.yml' which seems wrong if you assume alphabetical ordering.

I'd recommend to add a sorting option to ls to not depend on platform default for "ls"'s ordering.

Comment (rhuss) [reply deploy/resources.go]: It's [here](https://github.com/apache/camel-k/blob/059ae48ac293c6d28783cc958d0c39cc505e26fc/script/embed_resources.sh#L41) where I would add a `| sort` in between.

Comment (lburgazzoli) [reply deploy/resources.go]: not sure what appended but I see this commit

    https://github.com/apache/camel-k/commit/c2053bfe2155bb1e418457840434d8b07bde6985

Comment (rhuss) [reply deploy/resources.go]: Sorry, there is already a sort at the end.

So no idea how this file was regenerated the last time.

Comment (lburgazzoli) [reply deploy/resources.go]: so looks like there are some differences between linux and osx implementations

**Comment by rhuss** [comment]
Test failure is unrelated to this change:

```
The job exceeded the maximum log length, and has been terminated.
```

**Comment by lburgazzoli** [comment]
working on it

File: deploy/resources.go (hunk 2 above)
Comment (rhuss) [reply deploy/resources.go]: I don't think so as this final `| sort` works the same on linux and osx. No idea though why the resource list checked in in master is not sorted.

I suggest to merge this in (as its sorted now), and check for a PR which changes this again into a non-sorted order.

File: pkg/cmd/install.go [hunk 3]
```diff
@@ -230,3 +236,49 @@ func (o *installCmdOptions) waitForPlatformReady(platform *v1alpha1.IntegrationP
...
+		result = multierr.Append(result, err)
+	}
+	return result
+}
+
+func errorIfContextIsNotAvailable(schema *runtime.Scheme, context string) error {
+	
+	if context == platform.NoContext {
+		// Indicates that nothing should be installed
```
Comment (rhuss) [inline pkg/cmd/install.go]: @lburgazzoli should we check also when "none" + another context is given, that this is an error, too ? (as according to the code 'none' takes precedences of anything else ?)

Comment (lburgazzoli) [reply pkg/cmd/install.go]: yes we should

Comment (rhuss) [reply pkg/cmd/install.go]: Ok, I will add.

**Comment by rhuss** [comment]
Updated with extra check.

File: pkg/cmd/install.go (hunk 3 above)
Comment (rhuss) [reply pkg/cmd/install.go]: added

**Pull request merged**